RossBlassingame/JPL-CUSeniorProjects
# Feature/mars 112 limited astar

What changes were made:
- Implemented Limited A*
- Split Greedy into Unlimited and Limited variants
- Split tests into separate suites for readability
- Moved GreedyAlgorithm.processSlope logic into MarsRover.canTraverse so that all algorithms use the
new slope calculation
- Added tests

What are the relevant JIRA tickets:
MARS-105, MARS-112

How can these changes be tested(test #?):
AlgorithmTest.testAstarAlgorithmLimited* tests (all currently passing)
AlgorithmTest.testGreedyAlgorithm* tests (all currently passing)
See also images in Slack

Comments:
Regarding the canTraverse update, if we're not ready to do this change, I can change it back. The change was made because discussions from meetings as well as a want to test A* limited with more reasonable (sub 80 degree) maximum slopes. All the images I sent to Slack were produced using the new canTraverse.
